src/core/midi/MidiClient.cpp

```diff
@@ -221,10 +221,12 @@ void MidiClientRaw::parseData( const unsigned char c )
 		case MidiNoteOff:
 		case MidiNoteOn:
 		case MidiKeyPressure:
+			// set velocity of key
+			m_midiParseData.m_midiEvent.setVelocity(m_midiParseData.m_buffer[1]);
 		case MidiChannelPressure:
 		case MidiProgramChange:
+			// set key (or for ChannelPressure/ProgramChange: set 1st param)
 			m_midiParseData.m_midiEvent.setKey(m_midiParseData.m_buffer[0]);
```

Comment:
I think `m_midiParseData.m_midiEvent.setParam(0, m_midiParseData.m_buffer[0]);` is semantically more correct, but I'll leave the decision up to you.